### PR TITLE
Rename from mer to sailfish

### PR DESCRIPTION
**Pull request by pvuorela**
Adjusted the setup so that it doesn't generate the docs twice, first on build time and then on install time.

@vigejolla @jpetrell @rainemak

### CODE REVIEW

File: config/sailfish-qdoc-template.prf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+docs.CONFIG += no_check_exist no_default_install
+
+docs_install.files = $${QCHFILE}
+docs_install.path = $${SAILFISH_QDOC.path}
+docs_install.CONFIG += no_check_exist no_build
+
+QMAKE_EXTRA_TARGETS += docs
+PRE_TARGETDEPS += docs
+
+INSTALLS += docs_install
```
Comment (martyone) [inline config/sailfish-qdoc-template.prf:32]: I would prefer not implying those. It's common that docs building doesn't come into the way of regular edit-build-test cycle, i.e., that the docs+install_docs targets need to be required explicitly.

Comment (pvuorela) [reply config/sailfish-qdoc-template.prf:32]: Disagree. Qdoc processes the documentation we have nearly instantly and consider it unlikely that many would be actually manually modifying the build system to save a second or two. If they want, it's likely still pretty easy by commenting out the doc subdir in the projects or something like that.

In the general case don't see why the documentation should get any special treatment from the other sub-targets. We don't do make translations & make translations_install, or plugins etc. And the point of this kind of templates is to remove repetitive content.

If this starts to matter in some project, it should be easy later on to add e.g. a different config option as a variant that requires explicit build & install. Or the projects can introduce their own CONFIG whether the doc subdirectory gets included in the build.

File: config/sailfish-qdoc-template.prf [hunk 2]
```diff
@@ -16,19 +16,19 @@ ENV += \
     BUILD_DIRECTORY=$${OUT_PWD} \
     TEMPLATE=$${SAILFISH_QDOC.template}
 
-qch_docs.target = $${QCHFILE}
-qch_docs.depends += FORCE
-qch_docs.commands = (env $$ENV $$QDOC $$QDOCCONF) && \
+docs.input = ..
+docs.output = $${QCHFILE}
+docs.commands = (env $$ENV $$QDOC $$QDOCCONF) && \
                     ($${QHELPGENERATOR} $${QHELPFILE} -o $${QCHFILE})
```
[the remark this answers is not in the document]
Comment (pvuorela) [reply config/sailfish-qdoc-template.prf]: Adjusted.

File: doc/sailfish-qdoc-template.qdocconf [hunk 3]
```diff
@@ -18,7 +18,7 @@ qhp.SailfishQDocTemplate.filterAttributes = sailfish-qdoc-template 1.0
 
 HTML.footer += \
     "<div class=\"footer\">\n" \
-    "  <p><acronym title=\"Copyright\">&copy;</acronym> 2015 Jolla OY</p>" \
+    "  <p><acronym title=\"Copyright\">&copy;</acronym> 2021 Jolla Oy</p>" \
```
[the remark this answers is not in the document]
Comment (rainemak) [reply doc/sailfish-qdoc-template.qdocconf:21]: But our official copyright header for source code is Ltd.

[the remark this answers is not in the document]
Comment (pvuorela) [reply doc/sailfish-qdoc-template.qdocconf:21]: That's some precedence, but technically this is not source code. De facto the docs seem to be this way now. Another question is does this even make sense on the resulting files, year destined to be outdated soon. Mostly just don't want to get too much stuck on these details on updating the template naming.

**Pull request merged**